### PR TITLE
Project custom fields

### PR DESCRIPTION
**Pull request by yashhjaggi1998**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct]https://github.com/checkmarx-ltd/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/checkmarx-ltd/open-source-template/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentaiton is a Wiki Update, please indicate desired changes within PR MD Comment*
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: src/services/clients/cxClient.ts [hunk 1]
```diff
@@ -156,6 +159,49 @@ export class CxClient {
         return scanResult;
     }
 
+    async updateProjectCustomFields(): Promise<void> {
+        this.log.info("Updating Project Custom Fields.");
+        let projectId = this.projectId;
+        let path = `projects/${projectId}`;
+
+        //reading project custom fields entered by user on UI.
+        const projectCustomFields = this.sastConfig.projectCustomFields.split(",");
```
Comment (umeshwaghode) [inline src/services/clients/cxClient.ts:168]: What happens when there is no comma and single key:value

File: src/services/clients/cxClient.ts [hunk 2]
```diff
@@ -156,6 +159,49 @@ export class CxClient {
...
+        const projectCustomFieldsIds = new Array<string>(projectCustomFields.length);
+        const projectCustomFieldsKeys = new Array<string>(projectCustomFields.length);
+        const projectCustomFieldsValues = new Array<string>(projectCustomFields.length);
+        for (let i = 0; i < projectCustomFields.length; i++) {
+            projectCustomFieldsKeys[i] = projectCustomFields[i].split(":")[0];
```
Comment (umeshwaghode) [inline src/services/clients/cxClient.ts:173]: What if key value delimeter used is not colon character?

File: src/services/clients/cxClient.ts [hunk 3]
```diff
@@ -156,6 +159,49 @@ export class CxClient {
...
+            projectCustomFieldsValues[i] = projectCustomFields[i].split(":")[1];
+        }
+
+        //reading project custom fields stored in SAST Portal
+        const fetchSASTProjectCustomFields = await this.httpClient.getRequest('customFields',{});
+        for (let i = 0; i < projectCustomFieldsKeys.length; i++){
+            for(let fetchSASTProjectCustomField of fetchSASTProjectCustomFields){
+                if(projectCustomFieldsKeys[i] === fetchSASTProjectCustomField.name){
```
Comment (umeshwaghode) [inline src/services/clients/cxClient.ts:181]: What if projectCustomFieldsKeys[i] has leading or trailing space characters

**Review by umeshwaghode — CHANGES_REQUESTED** [review]

File: src/services/clients/cxClient.ts (hunk 2 above)
Comment (yashhjaggi1998) [reply src/services/clients/cxClient.ts:173]: These are input checks that need to be performed on VSTS code.

File: src/services/clients/cxClient.ts (hunk 3 above)
Comment (yashhjaggi1998) [reply src/services/clients/cxClient.ts:181]: It does not have it. This is the payload response from server so it has defined structure.

**Review by umeshwaghode — APPROVED** [review]

**Review by sangamchitmugre — APPROVED** [review]

**Pull request merged**